modules/secrets-store-azure.adoc

```diff
@@ -0,0 +1,275 @@
+  kind: ClusterRole
+  name: csi-secrets-store-provider-azure-cluster-role
+subjects:
+- kind: ServiceAccount
+  name: csi-secrets-store-provider-azure
+  namespace: openshift-cluster-csi-drivers
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  namespace: openshift-cluster-csi-drivers
+  name: csi-secrets-store-provider-azure
+  labels:
+    app: csi-secrets-store-provider-azure
+spec:
+  updateStrategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app: csi-secrets-store-provider-azure
+  template:
+    metadata:
+      labels:
+        app: csi-secrets-store-provider-azure
+    spec:
+      serviceAccountName: csi-secrets-store-provider-azure
+      hostNetwork: true
+      containers:
+        - name: provider-azure-installer
+          image: mcr.microsoft.com/oss/azure/secrets-store/provider-azure:v1.4.1
+          imagePullPolicy: IfNotPresent
+          args:
+            - --endpoint=unix:///provider/azure.sock
+            - --construct-pem-chain=true
+            - --healthz-port=8989
+            - --healthz-path=/healthz
+            - --healthz-timeout=5s
+          livenessProbe:
+            httpGet:
+              path: /healthz
+              port: 8989
+            failureThreshold: 3
+            initialDelaySeconds: 5
+            timeoutSeconds: 10
+            periodSeconds: 30
+          resources:
+            requests:
+              cpu: 50m
+              memory: 100Mi
+            limits:
+              cpu: 50m
+              memory: 100Mi
+          securityContext:
+            privileged: true
+          volumeMounts:
+            - mountPath: "/provider"
+              name: providervol
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: type
+                operator: NotIn
+                values:
+                - virtual-kubelet
+      volumes:
+        - name: providervol
+          hostPath:
+            path: "/var/run/secrets-store-csi-providers"
+      tolerations:
+      - operator: Exists
+      nodeSelector:
+        kubernetes.io/os: linux
+----
++
+[NOTE]
+====
+This configuration is modified from the configuration provided in the link:https://azure.github.io/secrets-store-csi-driver-provider-azure/docs/getting-started/installation/[Azure documentation] so that it works properly with {product-title}.
+====
++
+////
+TODO: Modified the above YAML with the following:
+
+- Added the ClusterRole and ClusterRoleBinding
+- Added namespace to the serviceaccount and daemonset
+- Changed securityContext to privileged on the container
+
+See if all of these are correct or not. Removed the "oc adm policy add-scc-to-use" step
+////
+
+.. Create the provider resources by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f azure-provider.yaml
+----
+
+. Create a secret provider class to define your secrets store provider:
```

Comment:
Hi @bergerhoffer in Azure platform, if we do not create secret generic secrets-store-creds in ns  "my-namespace" before creation of secretproviderclass and deployment we get error as 
```suggestion
. Warning  FailedMount  12s (x6 over 27s)  kubelet            MountVolume.SetUp failed for volume "secrets-store-inline" : fetching NodePublishSecretRef testropatil/secrets-store-creds failed: kubernetes.io/csi: failed to find the secret secrets-store-creds in the namespace testropatil with error: secrets "secrets-store-creds" not found
```

I guess we need to create secret in azure provider.
SERVICE_PRINCIPAL_CLIENT_SECRET="$(az ad sp create-for-rbac --name https://$KEYVAULT_NAME --query 'password' -otsv)"
SERVICE_PRINCIPAL_CLIENT_ID="$(az ad sp list --display-name https://$KEYVAULT_NAME --query '[0].appId' -otsv)"

oc create secret generic secrets-store-creds -n my-namespace --from-literal clientid=${SERVICE_PRINCIPAL_CLIENT_ID} --from-literal clientsecret=${SERVICE_PRINCIPAL_CLIENT_SECRET}
oc -n my-namespace label secret secrets-store-creds secrets-store.csi.k8s.io/used=true
cc: @dobsonj

Reply:
@ropatil010 So are these steps specific to Azure (and not needed by AWS)?

And can you give me a little more information on what each of these commands is doing, just so I can describe them properly in the steps

Reply:
Hi @bergerhoffer 
Line no 163: Create a Service Principal for the keyvault 
Which includes only these 2 commands from line no: 165 to Line no 177